Client/WarFare/UICmd.cpp

```diff
@@ -56,31 +56,31 @@ CUICmd::~CUICmd()
 bool CUICmd::Load(HANDLE hFile)
 {
 	if(CN3UIBase::Load(hFile)==false) return false;
-
-	m_pBtn_Act =			GetChildByID("btn_control");	//__ASSERT(m_pBtn_Act, "NULL UI Component!!");
-	m_pBtn_Act_Walk =		GetChildByID("btn_walk");		//__ASSERT(m_pBtn_Act_Walk, "NULL UI Component!!");
-	m_pBtn_Act_Run =		GetChildByID("btn_run");		//__ASSERT(m_pBtn_Act_Run, "NULL UI Component!!");
-	m_pBtn_Act_Stop =		GetChildByID("btn_stop");		//__ASSERT(m_pBtn_Act_Stop, "NULL UI Component!!");
-	m_pBtn_Act_Attack =		GetChildByID("btn_attack");		//__ASSERT(m_pBtn_Act_Attack, "NULL UI Component!!");
-	m_pBtn_Act_StandUp =	GetChildByID("btn_stand");		//__ASSERT(m_pBtn_Act_StandUp, "NULL UI Component!!");
-	m_pBtn_Act_SitDown =	GetChildByID("btn_sit");		//__ASSERT(m_pBtn_Act_SitDown, "NULL UI Component!!");
-
-	if(m_pBtn_Act_StandUp) m_pBtn_Act_StandUp->SetVisible(false); // 일어서기 버튼은 미리 죽여놓는다..
 	
-	m_pBtn_Character =		GetChildByID("btn_character");	//__ASSERT(m_pBtn_Character, "NULL UI Component!!");
-	m_pBtn_Inventory =		GetChildByID("btn_inventory");	//__ASSERT(m_pBtn_Inventory, "NULL UI Component!!");
-
-	m_pBtn_CmdList =			GetChildByID("btn_option");		//__ASSERT(m_pBtn_Option, "NULL UI Component!!");
-	m_pBtn_Camera =			GetChildByID("btn_camera");		//__ASSERT(m_pBtn_Camera, "NULL UI Component!!");
-	m_pBtn_Party_Invite =	GetChildByID("btn_invite");		//__ASSERT(m_pBtn_Party_Invite, "NULL UI Component!!");
-	m_pBtn_Party_Disband =	GetChildByID("btn_disband");	//__ASSERT(m_pBtn_Party_Disband, "NULL UI Component!!");
-	m_pBtn_Skill =			GetChildByID("btn_skill");		//__ASSERT(m_pBtn_Skill, "NULL UI Component!!");
-	m_pBtn_Exit	=			GetChildByID("btn_Exit");		//__ASSERT(m_pBtn_Exit, "NULL UI Component!!");
-
-	m_pBtn_Quest =		GetChildByID("btn_quest");			//__ASSERT(m_pBtn_Quest, "NULL UI Component!!");
-	m_pBtn_Belong =		GetChildByID("btn_knight");			//__ASSERT(m_pBtn_Belong, "NULL UI Component!!");
+	N3_VERIFY_UI_COMPONENT(m_pBtn_Act, GetChildByID("btn_control"));
```

Comment:
Preserve indentation after the comma.

Reply:
To answer your question though, typically we use the explicit type so it's clear.
Any other behaviour is just old outdated mgame behaviour. But you just have to make sure that it is in fact the suggested type before changing it, just in case.

Note that I didn't go through and mass-replace these because the implementation is still iffy.

Ideally we should double-check the type is valid, rather than just blindly casting and hoping for the best, which the macro doesn't really allow for.

Reply:
I've just merged in template specializations for this. So they can now just use GetChildByID<type> for the specific types (don't bother supplying a type if you just want CN3UIBase).

These are more strictly type checked and the change fits with the existing macros.